### ISSUE TITLE
pattern= should be a configurable setting in lint_dir()

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2573]
In {data.table}, for historical reasons we use the extension .Rraw for test files, thus [we have to run](https://github.com/Rdatatable/data.table/blob/014c82fd7b04e6e41da9680d41b25620a11f46b6/.github/workflows/lint.yaml#L31):

```r
lintr::lint_package(pattern = "(?i)[.](r|rmd|rraw)$")
```

(we don't use any of the other extensions in the default `pattern=` so we exclude them for simplicity)

This is fine in CI, but makes it hard to run interactively as we have to remember to add a strange regex to every execution of `lint_package()`.

Instead it would make sense if `pattern=` could be set by the config -- this is something that is reasonable to be fixed at a repo level and would allow the easy-to-remember approach of "always run `lint_package()`" to apply even locally.
